### PR TITLE
providing support for life cycle permissions and validations

### PR DESCRIPTION
**Pull request by sameerak**
[1] https://wso2.org/jira/browse/STORE-1126

### CODE REVIEW

File: apps/publisher/modules/lifecycle/lifecycle-api.js [hunk 1]
```diff
@@ -316,6 +374,7 @@ var error = '';
         var lifecycle = lcApi.getLifecycle(lcName, tenantId);
         var rxtManager = coreApi.rxtManager(tenantId);
         var lcState = am.getLifecycleState(asset, lcName);
+		var lcCheckedStates = am.getLifecycleCheckedState(asset.id, lcName);
```
Comment (manuranga) [inline apps/publisher/modules/lifecycle/lifecycle-api.js]: formatting issue

File: apps/publisher/modules/lifecycle/lifecycle-api.js [hunk 2]
```diff
@@ -334,7 +393,9 @@ var error = '';
             state.isDeletable = isDeletable(lcState, state.deletableStates, lcName);
         }
         //Update the state of the check items
-        state.checkItems = setCheckItemState(state.checkItems, asset, am, lcName);
+        state.checkItems = setCurrentCheckItemState(state.checkItems, lcCheckedStates);
+        state.approvedActions = new Object();
```
Comment (manuranga) [inline apps/publisher/modules/lifecycle/lifecycle-api.js]: don't do new Object, use state.approvedActions = {};

File: apps/publisher/themes/default/js/lifecycle/lifecycle-core.js [hunk 3]
```diff
@@ -514,10 +520,25 @@ var LifecycleUtils = {};
         var transition;
         for (var index = 0; index < transitions.length; index++) {
             transition = transitions[index];
-            actions.push(transition.event);
+            if (currentState == this.currentState) {
+                if (state.allowedActions && state.allowedActions[transition.event]){
```
Comment (manuranga) [inline apps/publisher/themes/default/js/lifecycle/lifecycle-core.js]: if's can be combined.

Comment (sameerak) [inline apps/publisher/themes/default/js/lifecycle/lifecycle-core.js]: If I combine these ifs UI will render all available actions of current state as allowed actions, even though none of the available actions are allowed for the current role.
Therefore it is necessary to split above if statements as they are added.
This way it is ensured that user who does not have privilege to perform any of the life-cycle transition actions, will not see actions buttons.

File: apps/publisher/modules/lifecycle/lifecycle-api.js (hunk 2 above)
Comment (sameerak) [inline apps/publisher/modules/lifecycle/lifecycle-api.js]: Fixed in commit 86e200939329fb59ede32b4bcdea56a8a9db2abd [1]

[1] https://github.com/sameerak/carbon-store/commit/86e200939329fb59ede32b4bcdea56a8a9db2abd

File: apps/publisher/modules/lifecycle/lifecycle-api.js (hunk 1 above)
Comment (sameerak) [inline apps/publisher/modules/lifecycle/lifecycle-api.js]: Fixed in commit 86e200939329fb59ede32b4bcdea56a8a9db2abd [1]

[1] https://github.com/sameerak/carbon-store/commit/86e200939329fb59ede32b4bcdea56a8a9db2abd

**Pull request merged**